File: apps/api/internal/service/domain/user/account.go
package user

import (
	"context"

	"fundlevel/internal/entities/account"
	"fundlevel/internal/storage"

	"github.com/google/uuid"
)

type UserService struct {
	repositories storage.Repository
}

func NewUserService(repositories storage.Repository) *UserService {
	return &UserService{
		repositories: repositories,
	}
}
func (s *UserService) GetAccount(ctx context.Context, userId uuid.UUID) (account.Account, error) {
	return s.repositories.User().GetAccount(ctx, userId)
}

File: apps/hono/src/internal/storage/implementaions/account.ts
import type { Account, CreateAccount } from "../../../internal/entities";
import { IAccountRepository } from "../../../internal/storage";
import { UUID } from "crypto";
import { SupabaseClient } from "@supabase/supabase-js";
import { Client, Database } from "@fundlevel/supabase/types";

export class AccountRepository implements IAccountRepository {
  constructor(private readonly sb: Client) { }

  async getByUserId(id: UUID): Promise<Account | undefined> {
    return (
      await this.sb.from("accounts").select("*").eq("user_id", id)
    ).data?.[0];
  }

  async create(params: CreateAccount): Promise<Account> {
    const { data, error } = await this.sb.from("accounts").insert(params).select().single();

    if (error || !data) {
      throw new Error("Failed to create account");
    }

    return data;
  }
}
